parser.go

```diff
@@ -190,11 +191,39 @@ func (p *parser) parseField() (*Field, error) {
 		}
 	}
 
+	var constrainedBy expression
```

Comment:
Here, you should consider using `peekWithChoice` to choose between `constrained_by` and `computed_by`, especially since the rest of the parsing -- i.e. `{`, `expr`, `}` -- is the same. You can reuse much of the code, and it will having to deal with having both a constraint and a computed by.

Fine to leave as is too.